test/unit/org/apache/cassandra/index/sai/cql/RandomItersectionTest.java

```diff
@@ -0,0 +1,209 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.index.sai.cql;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import org.apache.cassandra.index.sai.utils.SAIRandomizedTester;
+
+public class RandomItersectionTest extends SAIRandomizedTester
```

Comment:
```suggestion
public class RandomIntersectionTest extends SAIRandomizedTester
```
Although I guess it's kind of a fun combination of iterator and intersection :p

(I have added value to this patch...my work is done...)

Reply:
This is really a great, compact (after a couple cleanups), powerful test. +100

Reply:
Dyslexia is a wonderful thing at times. I usually rely on Intellij and the like to pick up these sort of things but it ignores names like this :)